src/api/spec/README.md

```diff
@@ -1,188 +1,117 @@
+```shell
+bundle exec rspec spec/models/user_spec.rb:33
 ```
 
-Please also have a look at the [Factory Bot documentation](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations).
+### Configuration
 
-#### Use a sequence for unique values
+- To extend the suite with custom configuration or custom methods we use ruby modules in `spec/support`
+- To configure how it's run by default we use an [.rspec](https://github.com/openSUSE/open-build-service/blob/master/src/api/.rspec) file.
 
-It's necessary to use a [sequence](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#sequences) for attributes which have to be unique like project.title or user.login.
+## What to test and how
 
-```ruby
-sequence(:login) { |n| "user_#{n}" }
-```
+You *can* test a lot of different things in a Ruby on Rails test suite. From behavior of units (as in methods of a Ruby object), to interactions between different units end-to-end (as in plugging different Ruby objects together to provide a feature. Like testing that `ActionDispatch`+`ActionController`+`ActiveRecord`+`ActionView` behave in a certain way from a browser) to accessibility testing or even more esoteric things like visual testing.
 
-Please keep in mind that you have to overwrite these attributes if they are part of the URI and you use it in combination with VCR.
-Otherwise, your tests will fail as VCR matches the cassette by the URI.
+But as we all know, too many cooks spoil the broth! That is why we are trying really hard to:
 
-```
-let!(:user) { create(:confirmed_user, login: "proxy_user") }
-```
-
-By passing ```login: "proxy_user"``` to the create statement, the username is now always proxy_user and not random (e.g. user_42).
-
-#### Factories should be the bare minimum
+- limit the amount of code involved in running the test
+- limit the amount of times we test the same code
+- limit the specs to "custom" logic we wrote ourselves
 
-Different to fixtures, Factory Bot runs through your ActiveRecord validations.
-That said, only add the bare minimum to your factory which is required to be valid.
-You can use an inherited factory to add or override attributes.
-
-```
-  factory :user do
-    email { Faker::Internet.email }
-    realname { Faker::Name.name }
-    sequence(:login) { |n| "user_#{n}" }
-    password 'buildservice'
-
-    factory :confirmed_user do
-      state 2
-    end
-```
+First and formost because a test suite is code that needs to be maintained and that is work, a lot of work. So writing the right type of spec, that tests *once* the code we have influence on, means *less* specs. Less specs, less code, less time spent on maintaining it. Secondly we think that CI time is developer time. A fast feedback loop from the test suite makes you wait for it less. So with those limitations we are trying to save the most precious thing we have as a project: The time you developer have, to work on something.
```

Comment:
```suggestion
First and foremost because a test suite is code that needs to be maintained and that is work, a lot of work. So writing the right type of spec, that tests *once* the code we have influence on, means *less* specs. Less specs, less code, less time spent on maintaining it. Secondly we think that CI time is developer time. A fast feedback loop from the test suite makes you wait for it less. So with those limitations we are trying to save the most precious thing we have as a project: The time you developer have, to work on something.
```